src/main/scala/beam/utils/mapsapi/googleapi/GoogleAdapter.scala

```diff
@@ -47,6 +77,13 @@ class GoogleAdapter(apiKey: String, outputResponseToFile: Option[Path] = None) e
     }
   }
 
+  private def parseResponse(response: HttpResponse) = {
+    val reduced = response.entity.dataBytes.runReduce(_ ++ _)
+    reduced.map { bs =>
+      Json.parse(bs.iterator.asInputStream).as[JsObject]
+    }
+  }
+
   private def parseRoutes(jsRoutes: Seq[JsValue]): Seq[Route] = {
```

Comment:
In case if API key is invalid, Goolge still gives back 200 with the following body:
```json
{
   "error_message" : "The provided API key is invalid.",
   "routes" : [],
   "status" : "REQUEST_DENIED"
}
```
Let's add some handing in case if `status` is not `OK`